### DISCUSSION TITLE
how to scroll a ScrollView to the bottom, with rust or slint?

### DISCUSSION THREAD

**Discussion by Luckie333** [slint-ui/slint#3405] [Q&A]
I have a ScrollView that keeps adding elements at runtime, how do I get it to scroll to the bottom with rust or slint code to make the last element visible.

**Comment by ogoffart** [comment]
You can change the viewport-y property like so: ` sv.viewport-y = - (sv.viewport-height - sv.visible-height);`

[Example](https://slintpad.com/?snippet=import+%7B+Button%2C+VerticalBox+%2C+AboutSlint%2C+ScrollView%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++width%3A+300px%3B%0A++++height%3A+300px%3B%0A++++sv+%3A%3D+ScrollView+%7B%0A++++++++VerticalLayout+%7B%0A++++++++++++Button+%7B%0A++++++++++++++++text%3A+%22Go+Down%22%3B%0A++++++++++++++++clicked+%3D%3E+%7B%0A++++++++++++++++++++sv.viewport-y+%3D+-+%28sv.viewport-height+-+sv.visible-height%29%3B%0A++++++++++++++++%7D+%0A++++++++++++%7D%0A++++++++++++Rectangle+%7B%0A++++++++++++++++height%3A+1000px%3B%0A++++++++++++++++background%3A+%40linear-gradient%280deg%2C+red%2C+blue%29%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++++++%0A++++%7D%0A%7D%0A)

(Note: I've moved this issue to Discussions as it's better suited for general questions and community interactions. This helps keep our issue tracker focused on bug reports and feature requests.)
